File: javascript/bubble_sort.js
function bubbleSort(arr) {
    const n = arr.length;
    for (let i = 0; i < n - 1; i++) {
        for (let j = i + 1; j < n; j++) {
            if (arr[i] > arr[j]) {
                [arr[i], arr[j]] = [arr[j], arr[i]];
            }
        }
    }
}

function generateArray(n) {
    // Simple seeded random for reproducible results
    let seed = 42;
    function seededRandom() {
        seed = (seed * 9301 + 49297) % 233280;
        return seed / 233280;
    }
    
    const arr = [];
    for (let i = 0; i < n; i++) {
        arr.push(Math.floor(seededRandom() * 10000));
    }
    return arr;
}

const args = process.argv.slice(2);
if (args.length < 1) {
    console.log('Usage: bubble_sort <n>');
    process.exit(1);
}

const n = parseInt(args[0]);
if (isNaN(n)) {
    console.log('Invalid number:', args[0]);
    process.exit(1);
}

const arr = generateArray(n);

const start = Date.now();
bubbleSort(arr);
const end = Date.now();

const timeMs = end - start;

console.log(`JavaScript: bubble_sort(${n}) = sorted`);
console.log(`Time: ${timeMs}ms`);

File: lua/bubble_sort.lua
function bubble_sort(arr)
    local n = #arr
    for i = 1, n - 1 do
        for j = 1, n - i do
            if arr[j] > arr[j + 1] then
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
            end
        end
    end
end

function generate_array(n)
    math.randomseed(42) -- Fixed seed for reproducible results
    local arr = {}
    for i = 1, n do
        arr[i] = math.random(0, 9999)
    end
    return arr
end

if #arg < 1 then
    print('Usage: bubble_sort <n>')
    os.exit(1)
end

local n = tonumber(arg[1])
if not n then
    print('Invalid number: ' .. arg[1])
    os.exit(1)
end

local arr = generate_array(n)

local start = os.clock()
bubble_sort(arr)
local finish = os.clock()

local time_ms = math.floor((finish - start) * 1000)

print('Lua: bubble_sort(' .. n .. ') = sorted')
print('Time: ' .. time_ms .. 'ms')